paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetTest.kt

```diff
@@ -0,0 +1,195 @@
+            "Must call `configure` first before attempting to fetch the saved payment option!"
+        )
+    }
+
+    @Test
+    fun `When Google payment option, should return option is config enables Google Pay`() = runPaymentOptionTest(
+        paymentOption = CustomerAdapter.PaymentOption.GooglePay,
+        configuration = CustomerSheet.Configuration.builder(merchantDisplayName = "Merchant, Inc.")
+            .googlePayEnabled(googlePayConfiguration = true)
+            .build(),
+    ) { result ->
+        val selectedResult = result.asSelected()
+
+        assertThat(selectedResult.selection).isInstanceOf(PaymentOptionSelection.GooglePay::class.java)
+    }
+
+    @Test
+    fun `When Google payment option, should not return option is config disables Google Pay`() = runPaymentOptionTest(
+        paymentOption = CustomerAdapter.PaymentOption.GooglePay,
+        configuration = CustomerSheet.Configuration.builder(merchantDisplayName = "Merchant, Inc.")
+            .googlePayEnabled(googlePayConfiguration = false)
+            .build(),
+    ) { result ->
+        val selectedResult = result.asSelected()
+
+        assertThat(selectedResult.selection).isNull()
+    }
+
+    @Test
+    fun `When Google payment option, should not return option is config has not Google Pay config`() =
+        runPaymentOptionTest(
+            paymentOption = CustomerAdapter.PaymentOption.GooglePay,
+            configuration = CustomerSheet.Configuration.builder(merchantDisplayName = "Merchant, Inc.")
+                .build(),
+        ) { result ->
+            val selectedResult = result.asSelected()
+
+            assertThat(selectedResult.selection).isNull()
+        }
+
+    private fun runPaymentOptionTest(
+        configuration: CustomerSheet.Configuration?,
+        paymentOption: CustomerAdapter.PaymentOption? = null,
+        test: (result: CustomerSheetResult) -> Unit,
+    ) {
+        runTestActivityTest { activity, complete ->
+            val customerSheet = CustomerSheet.create(
+                activity = activity,
+                customerAdapter = FakeCustomerAdapter(
+                    selectedPaymentOption = CustomerAdapter.Result.success(paymentOption),
+                ),
+                callback = {},
+            )
+
+            configuration?.let {
+                customerSheet.configure(configuration)
+            }
+
+            runBlocking {
+                test(customerSheet.retrievePaymentOptionSelection())
+
+                complete()
+            }
+        }
+    }
+
+    private fun runTestActivityTest(
+        test: (TestActivity, () -> Unit) -> Unit,
+    ) {
+        ActivityScenario.launch(TestActivity::class.java).use { scenario ->
+            val countDownLatch = CountDownLatch(1)
+
+            scenario.onActivity {
+                test(it) {
+                    countDownLatch.countDown()
+                }
+            }
+
+            countDownLatch.await()
```

Comment:
We should do an `assertThat(countDownLatch.await(1, TimeUnit.Secionds).isTrue()`, otherwise these can hang and be harder to debug.